expressEngine/user/service/user.js

```diff
@@ -88,6 +89,23 @@ class Service {
     }
     return ({ data: deleted, message: success.userDeleted });
   }
+
+  put = async ({ body }) => {
+    const { id, dataToUpdate } = body;
+    if (!dataToUpdate || Object.entries(dataToUpdate).length === 0) {
+      return { error: error.emptyData };
+    }
+    if (dataToUpdate.password) {
+      return { error: error.notUpdatable };
+    }
+    if (dataToUpdate.email && !modelLib.validateEmail(dataToUpdate.email)) {
+      return { error: `${dataToUpdate.email} is not a valid email!` };
+    }
+    const updated = await DBService.updateOne(User, { userId: id }, dataToUpdate);
+    return (updated.nModified
+      ? { message: success.userUpdated }
+      : { error: error.unableToUpdate });
```

Comment:
@adilsuccessive unableToUpdate error is not returning when the user enters the wrong field in dataToUpdate.

Reply:
@chiragSuccessive this should handle in the middleware in the parser to accept only predefined field of the schema.